shell/initialize/index.js

```diff
@@ -121,6 +123,12 @@ async function createApp(ssrContext, config = {}) {
   // Add this.$router into store actions/mutations
   store.$router = router;
 
+  // Add racher-api classes to Vue prototype
+  Vue.use(rancherApiPlugin, { store });
+  Vue.use(clusterApiPlugin, { store });
+  Vue.use(shellApiPlugin, { store });
+  Vue.use(extensionApiPlugin, { store });
```

Comment:
Will these make the API available in the bootstraping methods of the extensions? If so, isn't there a possibility that some of the stores won't be loaded yet? We might have to make that clear or error out gracefully if that's the case.

Reply:
This doesn't make the API available to extensions after some initial testing, I don't think the types are being properly referenced in an extension.

Perhaps creating a helper function to check for the existence of the store before running any commands wouldn't be a bad idea, but I believe the plugins (extensions) are initialized on line 324 which is after both the store and prototypes are created:
```js
  if (process.client && typeof plugin === 'function') {
    await plugin(app.context, inject);
  }
```